projects/elements/src/lib/lazy-elements/lazy-element/lazy-element.directive.ts

```diff
@@ -26,14 +32,19 @@ const LOG_PREFIX = '@angular-extensions/elements';
   selector: '[axLazyElement]',
 })
 export class LazyElementDirective implements OnInit, OnDestroy {
-  @Input('axLazyElement') url: string;
+  @Input('axLazyElement')
+  set url(url: string) {
+    this.url$.next(url);
```

Comment:
Could we do this just as a plain data and `ngOnChanges` instead?

I have pretty bad experience with the proposed pattern as it grows, single field is ok but why streamify something which is a static string?

Reply:
Yeah; sure.

Reply:
Sure, I'll switch to `ngOnChanges`.

The stream is used only in favor of cancellation purposes and to avoid data races, e.g. if the first `loadElement` promise resolves later than the second one, then it'll render the wrong element.

Reply:
@arturovt yeah ok I see your point, this is in fact async processing, then lets keep it like you proposed initially, sorry... Haven't look in depth just on surface ;)

Reply:
Of course robustness always wins vs code style